README.md

```diff
@@ -355,6 +356,38 @@ writable.
 
 Always prefer `ReadonlyArray` over a regular `Array` unless it must be possible to modify the Array.
 
+### React props & TypeScript
+
+`interface` is preferred over `type` when defining the component props. Are a little more powerful
+than the corresponding type declaration, but for a set of react props, it likely doesn't matter.
+
+Also is not necessary to type the `returnType` because is inferred from what you actually return, so
+can be omitted most of the time.
```

Comment:
Some typos:
* Also is not -> Also **it** is not
* because is inferred-> because **it** is inferred

Regarding the actual statement I agree and disagree at the same time, for 99% of the cases it's probably fine to leave the return type untyped and let TS infer it as `JSX.Element`. However, in the 1% of cases where you _might_ return something other than JSX, stuff can break.

I know that @mmnathan is hot on explicitly typing the return types.

Reply:
I'm with Nathan on that, public boundaries should be typed where possible to show clear intent.

It's a contract. Changing the implementation should never implicitly change the return type. When it's explicit, and you know you have to change the return type, you know you might need to refactor other code as well.

I know react components are bit different, since they have a well defined API, but it's still useful to spot errors in your code :)